database/seeders/DefaultSkills.php

```diff
@@ -12,50 +13,105 @@ class DefaultSkills extends Seeder
+    {
+        $raw = json_decode(file_get_contents($jsonPath), true);
+
+        if (!is_array($raw)) {
+            throw new \Exception("Invalid JSON in $jsonPath");
+        }
 
-            if (!is_array($raw)) {
-                throw new \Exception("Invalid JSON in $jsonPath");
+        $data = [];
+        foreach ($raw as $categoryId => $skills) {
+            foreach ($skills as $skill) {
+                $data[] = [
+                    'skill_name' => $skill['skill_name'],
+                    'category' => (int)$categoryId,
+                    'description' => $skill['description'] ?? ''
+                ];
             }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Get fallback skills data
+     */
+    private function getFallbackData(): array
+    {
+        return [
+            ['skill_name' => 'Publicising events', 'category' => 1],
+            ['skill_name' => 'Recruiting volunteers', 'category' => 1],
+            ['skill_name' => 'Managing events', 'category' => 1],
+            ['skill_name' => 'Finding venues', 'category' => 1],
+
+            ['skill_name' => 'Software/OS', 'category' => 2],
+            ['skill_name' => 'Changing a fuse', 'category' => 2],
+            ['skill_name' => 'Using a multimeter', 'category' => 2],
+            ['skill_name' => 'Laptop disassembly', 'category' => 2],
+            ['skill_name' => 'Replacing PCB components', 'category' => 2],
+            ['skill_name' => 'Headphones', 'category' => 2],
+            ['skill_name' => 'Electronics safety', 'category' => 2],
+            ['skill_name' => 'Replacing screens', 'category' => 2],
+        ];
+    }
+
+    /**
+     * Seed skills to database based on configuration
+     */
+    private function seedSkills(array $data): void
+    {
+        $truncate = env('SEEDING_TRUNCATE_SKILLS', true);
 
-            // Flatten the new schema: { <category_id>: [ { skill_name, description? }, ... ] }
-            $data = [];
-            foreach ($raw as $categoryId => $skills) {
-                foreach ($skills as $skill) {
-                    $entry = [
-                        'skill_name' => $skill['skill_name'],
-                        'category' => (int)$categoryId,
-                    ];
-                    if (isset($skill['description'])) {
-                        $entry['description'] = $skill['description'];
-                    }
-                    $data[] = $entry;
+        if ($truncate) {
+            Log::info('Truncating and re-inserting all skills');
+            DB::table('skills')->truncate();
+            DB::table('skills')->insert($data);
+            Log::info('Inserted ' . count($data) . ' skills');
+        } else {
+            Log::info('Merging new skills only');
+            $added = 0;
+            
+            foreach ($data as $entry) {
+                if (!DB::table('skills')->where('skill_name', $entry['skill_name'])->exists()) {
```

Comment:
Can different categories have the same `skill_name`? Do we want this to check categoryid and skill_name?

Reply:
Also, if the categoryid and skill_name are unchanged, but we want to update the description, this won't do that. Not sure if there's a use case for that.

Reply:
Addressed in [999901e](https://github.com/iFixit/restarters/pull/17/commits/999901e73a15493e104ce2aa20ad2ffdcabb4ebe)

Reply:
We will be able to update the skill description now, but we still only check the existence of a skill based on the name alone.

The only case where I think we'd have a skill in two different categories is if we had first placed it in the wrong category and then tried to correct it by placing it in the other. Otherwise, it does not make sense to have a skill in two different categories:

![SnagitHelper2023 2025-05-13 10 39 53](https://github.com/user-attachments/assets/d1eeee4d-269d-4dcc-ac72-329879288240)

In which case, we'd manually delete the skill from the wrong category.